webproject/src/main/java/com/google/sps/dataManagers/UserRoomManager.java

```diff
@@ -0,0 +1,65 @@
+package com.google.sps.dataManagers;
+
+import java.util.concurrent.CountDownLatch;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.ValueEventListener;
+import com.google.sps.data.UserRoom;
+import com.google.sps.util.FirebaseUtil;
+
+@Singleton
+public class UserRoomManager {
+    private FirebaseUtil firebaseUtil;
+
+    @Inject
+    public UserRoomManager(FirebaseUtil firebaseUtil) {
+        this.firebaseUtil = firebaseUtil;
+    }
+
+    public void addUserRoom(String userEmail, String roomId) {
+        UserRoom userRoom = UserRoom.newBuilder().setUserEmail(userEmail).setRoomId(roomId).setUserEmailRoom().build();
+
+        firebaseUtil.getUserRoomReference().push()
+            .setValue(userRoom, (databaseError, databaseReference) -> {
+                if (databaseError != null) {
+                    System.out.println("Data could not be saved " + databaseError.getMessage());
+                } else {
+                    System.out.println("Data saved successfully.");
+                }
+            });
+    }
+
+    public boolean hasUserJoinedRoom(String userEmail, String roomId) {
+        final boolean[] isUserInRoom = { false };
+        CountDownLatch done = new CountDownLatch(1);
+
+        firebaseUtil.getUserRoomReference().addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                for (DataSnapshot data : dataSnapshot.getChildren()) {
+                    if (data.child("userEmailRoom").getValue().equals(userEmail + "_" + roomId)) {
+                        isUserInRoom[0] = true;
+                    }
+                }
+                done.countDown();
+            }
+
+            @Override
+            public void onCancelled(DatabaseError error) {
```

Comment:
dont need to override this

Reply:
i need to override it, otherwise it gives an error